gpt_researcher/agent.py

```
@@ -370,43 +448,125 @@ async def _handle_deep_research(self, on_progress=None):
         # Return the research context
         return self.context
 
-    async def write_report(self, existing_headers: list = [], relevant_written_contents: list = [], ext_context=None, custom_prompt="") -> str:
+    async def write_report(
+        self,
+        existing_headers: list = [],
```

Ruff (0.15.5)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)